### PR TITLE
[Identity] `az identity federated-credential`: Add subgroup to support managing federated identity credentials of existing user assigned identities

### LINKED ISSUE #23152: Release of MSI version 2022-01-31-preview
### Related command

az identity

### Resource Provider

Microsoft.ManagedIdentity

### Description of Feature or Work Requested

MSI has added a new resource Federated Identity Credentials (FIC) as a sub-resource of existing User Assigned Identities "userAssignedIdentities/federatedIdentityCredentials"

CLI is expected to provide CRUD operation for the given sub-resource. Take a look Request example.

high-level documentation: https://aka.ms/ami/wif/docs

FIC object looks like this:
```
{
        "issuer": "https://oidc.prod-aks.azure.com/IssuerGUID",
        "subject": "system:serviceaccount:ns:svcaccount",
        "audiences": ["api://AzureADTokenExchange"],
}
```
In AAD context Audiences array is supposed to have exactly one element in most of the cases. 
Is it possible to make "audiences": ["api://AzureADTokenExchange"] a default value with the ability to override?

### Minimum API Version Required

2022-01-31-preview

### Swagger PR link

https://github.com/Azure/azure-rest-api-specs/pull/19548

#### Existing command to create managed identity
```
az identity create --name $uaId --resource-group $rg --location $location --subscription $subscription
```
then this managed identity can be used for FIC CRUD operations below. Those 4 operations are expected to become available in CLI.

#### create/update FIC
```
az rest --method put `
--url "/subscriptions/$subscription/resourceGroups/$rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/$uaId/federatedIdentityCredentials/$ficId?api-version=2022-01-31-PREVIEW" `
--headers "Content-Type=application/json" `
--body "{'properties': { 'issuer': 'https://kubernetes-oauth.azure.com/', 'subject': 'system:serviceaccount:ns:svcaccount', 'audiences': ['api://AzureADTokenExchange'] }}"
```

#### read FIC
```
az rest --method get --url "/subscriptions/$subscription/resourceGroups/$rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/$uaId/federatedIdentityCredentials/$ficId?api-version=2022-01-31-PREVIEW"
```

#### supports paging
```
az rest --method get --url "/subscriptions/$subscription/resourceGroups/$rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/$uaId/federatedIdentityCredentials?api-version=2022-01-31-PREVIEW"
```

#### delete fic
```
az rest --method delete --url "/subscriptions/$subscription/resourceGroups/$rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/$uaId/federatedIdentityCredentials/$ficId?api-version=2022-01-31-PREVIEW"
```

### Target Date

July

### Additional context

pypi link: https://pypi.org/project/azure-mgmt-msi/6.1.0/
Given API version "2022-01-31-PREVIEW" is available in public cloud, mooncake, fairfax, but not air-gapped clouds

### PR DESCRIPTION
**Pull request by yanzhudd**
**Related command**

create FIC:
az identity federated-credential create --name myFicName --identity-name myIdentityName --resource-group myResourceGroup --issuer myIssuer --subject mySubject --audiences myAudiences

update FIC:
az identity federated-credential update --name myFicName --identity-name myIdentityName --resource-group myResourceGroup --issuer myIssuer --subject mySubject --audiences myAudiences

read FIC:
az identity federated-credential show --name myFicName --identity-name myIdentityName --resource-group myResourceGroup

read all FICs associated with the user-assigned identity:
az identity federated-credential list --identity-name myIdentityName --resource-group myResourceGroup

delete FIC:
az identity federated-credential delete --name myFicName --identity-name myIdentityName --resource-group myResourceGroup

**Description**

In `az identity` group, add subgroup `az identity federated-credential` to support managing federated identity credentials of existing user assigned identities
Close #23152

**Testing Guide**

**History Notes**

[Compute] `az identity federated-credential`: Add subgroup to support managing federated identity credentials of existing user assigned identities

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/identity/commands.py [hunk 1]
```diff
@@ -31,3 +36,12 @@ def load_command_table(self, _):
...
     with self.command_group('identity', msi_operations_sdk, client_factory=_msi_operations_operations) as g:
         g.command('list-operations', 'list')
+
+    with self.command_group('identity federated-credential', federated_identity_credentials_sdk,
+                            client_factory=_msi_federated_identity_credentials_operations,
+                            min_api='2022-01-31-preview') as g:
+        g.custom_command('create', 'create_or_update_federated_credential')
+        g.custom_command('update', 'create_or_update_federated_credential')
+        g.custom_show_command('show', 'show_federated_credential')
+        g.custom_command('delete', 'delete_federated_credential')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/identity/commands.py:46]: Do we need to consider adding `confirmation=True` for the delete command?

Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/identity/commands.py:46]: Yes, adding a confirmation for `delete` command is better. I will add it.

**Review by zhoxing-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/identity/_help.py [hunk 2]
```diff
@@ -28,3 +30,53 @@
...
+"""
+
+helps['identity federated-credential create'] = """
+type: command
+short-summary: Create a federated identity credential under an existing user assigned identity.
+examples:
+  - name: Create a federated identity credential under a specific user assigned identity.
+    text: |
+        az identity federated-credential create --name myFicName --identity-name myIdentityName --resource-group myResourceGroup --issuer myIssuer --subject mySubject --audiences myAudiences
+"""
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/identity/_help.py:46]: This command `az identity federated-credential create` acts on **managed identity**: https://learn.microsoft.com/en-us/azure/active-directory/workload-identities/workload-identity-federation-create-trust-user-assigned-managed-identity?pivots=identity-wif-mi-methods-azcli

`az ad app federated-credential create` (#23122) acts on **AAD application**: https://learn.microsoft.com/en-us/azure/active-directory/workload-identities/workload-identity-federation-create-trust?pivots=identity-wif-apps-methods-azcli

However, they expose different command interfaces:

- `az identity federated-credential create` uses separated arguments `--issuer`, `--subject` and `--audiences`
- `az ad app federated-credential create` uses `--parameters` which includes `issuer`, `subject` and `audiences` in a JSON payload. This decision was made in email thread: _federatedIdentityCredentials's implementation in Azure CLI_

In my opinion, this causes inconsistency and increases users' learning cost.

**Pull request merged**